docs/assembly.rst

```diff
@@ -947,11 +980,14 @@ Pseudocode::
       POP
     LabelDefinition(name [id1, ..., idn] :) ->
       JUMPDEST
-      // register new variables id1, ..., idn and set the stack height to
-      // stack_height_at_block_start + number_of_local_variables
+      // register new variables id1, ..., idn and adjust the assumed stack
+      // height such that it matches the stack height at the beginning of
+      // the block plus all local variables including the just registered
```

Comment:
What about the "not-yet registered" local variables that appear in the later labels?  What about the "already-registered" local variables that appear in previous labels?